prov/gni/src/gnix_mr_cache.c

```diff
@@ -943,6 +1268,10 @@ static int __mr_cache_search_stale(
 	gnix_mr_cache_key_t *mr_key;
 	gnix_mr_cache_entry_t *mr_entry, *tmp;
 
+	if (cache->attr.lazy_deregistration) {
+		__clear_notifier_events(cache);
```

Comment:
Is the notifier only ever used in lazy deregistration? I'd assume this would affects all versions of the cache, since a region could be unmapped and the user could attempt to register a subsection of an existing registration that has been partially unmapped.